src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -535,7 +587,15 @@ ss::future<> partition_downloader::download_segment_file(
         co_await ss::remove_file(localpath.string());
     }
 
-    auto stream = [this, part, remote_path, _localpath{localpath}, _otl{otl}](
+    model::offset min_offset;
+    model::offset max_offset;
+    auto stream = [this,
+                   part,
+                   remote_path,
+                   _localpath{localpath},
+                   _otl{otl},
+                   &min_offset,
+                   &max_offset](
                     uint64_t len,
```

Comment:
I'm not sure but I think even pod captured values (such as `this`) must be copied to the stack because the lambda object ceases to exist after the first suspend point.

Reply:
I hadn't consider `this`, but other captures likely do. If `this` needs to be copied, then I wonder if even the captured references need to be as well.

Reply:
good catch, fixed that